frontend/src/ui-components/AcmSearchInput/AcmSearchInput.tsx

```diff
@@ -0,0 +1,273 @@
+    placeholder,
+    useAdvancedSearchPopper,
+    spellCheck,
+    resultsCount,
+    style,
+  } = props
+
+  const searchInputRef = useRef<HTMLInputElement | null>(null)
+  const advancedSearchPaneRef = useRef<HTMLDivElement | null>(null)
+  const [isAdvancedSearchOpen, setIsAdvancedSearchOpen] = useState(false)
+  const [searchConstraints, setSearchConstraints] = useState<SearchConstraint[]>(pendingConstraints || [])
+
+  const onClear = () => {
+    fuzzySearchOnClear && fuzzySearchOnClear()
+    setSearchConstraints([{ operator: SearchOperator.Equals, value: '', columnId: '' }])
+    setActiveConstraints && setActiveConstraints([{ operator: SearchOperator.Equals, value: '', columnId: '' }])
+  }
+
+  const onChange = (value: any) => {
+    fuzzySearchOnChange && fuzzySearchOnChange(value)
+  }
+
+  // clicking outside the menu should close it
+  const handleClickOutside = (event: any) => {
+    if (!isAdvancedSearchOpen || !advancedSearchPaneRef.current) return
+
+    const isClickOutside =
+      !searchInputRef.current?.contains(event.target) && !advancedSearchPaneRef.current.contains(event.target)
+
+    if (isClickOutside) {
+      setIsAdvancedSearchOpen(false)
+    }
+  }
+
+  useEffect(() => {
+    window.addEventListener('click', handleClickOutside)
+    return () => {
+      window.removeEventListener('click', handleClickOutside)
+    }
+  })
+
+  const getAvailableColumnNames = () => {
+    return searchableColumns?.map(({ columnId, columnDisplayName }) => {
+      const isDisabled = !!searchConstraints.find(
+        (constraint) => constraint.columnId === columnId && constraint.operator === SearchOperator.Equals
+      )
+      return (
+        <SelectOption
+          key={columnId}
+          value={columnId}
+          isDisabled={isDisabled}
+          onClick={(event) => {
+            event.stopPropagation()
+          }}
+        >
+          {columnDisplayName || columnId}
+        </SelectOption>
+      )
+    })
+  }
+
+  const activeConstraints = searchConstraints.filter(
+    (constraint) => constraint.columnId && constraint.value && constraint.operator
+  ).length
+  const showResultCount = !!fuzzySearchValue || !!activeConstraints
+  const searchInput = (
+    <SearchInput
+      placeholder={placeholder}
+      spellCheck={spellCheck}
+      resultsCount={resultsCount}
+      ref={searchInputRef}
+      id="custom-advanced-search"
+      style={style}
+      onChange={(_event, value) => onChange(value)}
+      onToggleAdvancedSearch={
+        useAdvancedSearchPopper
+          ? (e, isOpen) => {
+              e.stopPropagation()
+              setIsAdvancedSearchOpen(isOpen!)
+            }
+          : undefined
+      }
+      isAdvancedSearchOpen={isAdvancedSearchOpen}
+      onClear={onClear}
+      value={fuzzySearchValue}
+      enableUtilities={showResultCount}
+    />
+  )
+
+  const advancedSearchForm = (
+    <div ref={advancedSearchPaneRef} aria-label="Advanced search form">
```

Comment:
aria-label should be translated, although, I'm not sure that it is usual to have one on a div.